src/yang/ietf-tcp.yang

```diff
@@ -190,26 +182,17 @@ module ietf-tcp {
              that can be part of either the client or server.";
         }
         description
-          "Connection related parameters.";
+          "List of TCP connections with their parameters. The list 
+           is modelled as writeable, but implementations may not allow
+           creation of new TCP connections by adding entries to the list.
+           Furthermore, the behavior upon removal is implementation-specific.
+           Implementations may support closing or reseting a TCP connection 
+           upon an operation that removes the entry from the list.";
```

Comment:
Actually, parameters on the list may be read/write, such as the keep-alive parameters or possibly also TCP-AO ones. See below.

The big question is how adding and removing of entries from the list would work. Removal may be possible; then the TCP connection would be closed. Adding a TCP connection via the YANG model may only be possible in very special enviroments.

Some implementations may decide to implement the list read-only; this is the easiest approach as well. But alternatives could exist. IMHO write access should be allowed but not mandatory-to-implement.

Reply:
> Actually, parameters on the list may be read/write, such as the keep-alive parameters or possibly also TCP-AO ones. See below.

You are correct. Those parameters need to rw for applications to be able to configure those values.

> 
> The big question is how adding and removing of entries from the list would work. Removal may be possible; then the TCP connection would be closed. Adding a TCP connection via the YANG model may only be possible in very special enviroments.

I am not the expert here, but I would imagine that using a YANG model to add, or delete connections would be unusual. See below.

> 
> Some implementations may decide to implement the list read-only; this is the easiest approach as well. But alternatives could exist. IMHO write access should be allowed but not mandatory-to-implement.

Agree, but I think we could nuance this by saying in the draft which attributes are truly rw, and which ones are ro. Agree?

Reply:
You are right, *adding* connections via the YANG model would be unusual. However, theoretically, *deleting* connections would be doable. For instance, assume a Carrier-Grade NAT device that runs out of TCP port numbers. In that case, an NMS could decide to force closing some TCP connections out of the list of all connections. There will most likely better was to realize this instead of deleting list entries. But the key question is if this shall be excluded. IMHO this could be implementation-specific behavior.

I am not sure whether we really shall describe in the descriptions if an attribute is ro or rw. At the end of the day, it may anyway be implementation-specific whether rw access is possible. The simpler alternative would be to have a global statement that rw access to entries in the connection list is implementation specific.